connector/mqtt/vernemq/v2k-brigde/bin/scripts_tls/ejbca_client.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/sh
+
+#########################################################
+### Required Packages: openssl, curl, jq
+### Expected environment variables, example:
+#
+# export EJBCA_HOSTNAME='localhost'
+# export EJBCA_PORT='5583'
+# export STATIC_CERT='n'
+# export USE_VMQ_OPERATOR='n'
+# export HOSTNAME='broker'
+# export CRL_UPDATE_TIME='*/30 * * * *'
+# export BASE_DIR='/vernemq'
+# export CHECKEND_EXPIRATION_SEC='43200'
+# export CHECK_EXPIRATION_TIME='*/30 * * * *'
+# export CHECK_BROKER_CERT_REVOKED_TIME='*/30 * * * *'
+# 
+#########################################################
+
+
+########################################################
+
+BASE_DIR=${BASE_DIR:-"/v2k_bridge"}
+
+
+. "${BASE_DIR}"/bin/scripts_tls/_initVariables.sh
+
+
+_removeCRTDir()
+{
+  if [ -d "$certDir" ];
+  then
+    rm -rf "${certDir}"
+  fi
+}
+
+_createCRTDir()
+{
+  mkdir "${certDir}"
+  cd "${certDir}" || exit
+}
+
+_connectEJBCA()
+{
+  # Waiting for dojot MQTT broker for at most 3 minutes
+  START_TIME=$(date +'%s')
+  echo "Waiting for dojot EJBCA Broker fully start. Host '${EJBCA_HOSTNAME}', '${EJBCA_PORT}'..."
+  echo "Try to connect to dojot EJBCA Broker ... "
+  RESPONSE=$(curl --fail -s "${certEjbcaApiUrl}"/ejbca/version || echo "")
+  echo "$RESPONSE"
+  while [ -z "${RESPONSE}" ]; do
+      sleep 30
+      echo "Retry to connect to dojot EJBCA broker ... "
+      RESPONSE=$(curl --fail -s "${certEjbcaApiUrl}"/ejbca/version || echo "")
+
+      ELAPSED_TIME=$(($(date +'%s') - ${START_TIME}))
+      if [ ${ELAPSED_TIME} -gt 180 ]
+      then
+          echo "dojot EJBCA broker is taking too long to fully start. Exiting!"
+          exit 3
+      fi
+  done
+  echo "dojot EJBCA broker at host '${EJBCA_HOSTNAME}', port '${EJBCA_PORT}' fully started."
+
+  # give time for EJBCA fully started
+  sleep 5
+}
+
+##Generate key par (private and public key)
+_generateKeyPair()
+{
+    sh "${BASE_DIR}"/bin/scripts_tls/generateKeyPair.sh
```

Comment:
Is it really necessary to run this script inside another shell `sh`?

would using the command below produce the same result (with the advantage of not creating a new process for that)?

`source ${BASE_DIR}"/bin/scripts_tls/generateKeyPair.sh` 
or the shortcut
`. ${BASE_DIR}"/bin/scripts_tls/generateKeyPair.sh`

Reply:
its not necessary